java-checks/src/main/java/org/sonar/java/checks/CheckList.java

```diff
@@ -345,7 +344,6 @@ private CheckList() {
       CustomUnclosedResourcesCheck.class,
       DanglingElseStatementsCheck.class,
       DataEncryptionCheck.class,
-      DataHashingCheck.class,
```

Comment:
As discussed this morning, we can not drop this rule directly, issues currently raised should not disappear, we should go through a deprecation period. 

At the end:

- S4790 implementation will change (and it happens to be matching closely S2070), to report fewer issues. And it's perfectly fine, it's like any improvement of a rule.
- S2070 will be deprecated. The implementation will be closely related to the new S4790, and it's not a problem either.

@m-g-sonar Does this solution looks good to you? You mention the possibility to use @DeprecatedRuleKey, but I'm unsure to see what it can improve.